src/figuresClasses.ts

```diff
@@ -1,11 +1,78 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    protected a: number,
+    protected b: number,
+    protected c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    const longestSide = Math.max(a, b, c);
+    const perimeter = a + b + c;
 
-export class Triangle implements Figure {}
+    if (longestSide >= perimeter - longestSide) {
+      throw new Error(
+        'The longest side must be shorter than the sum of the other two sides',
+      );
```

Comment:
[CODE STYLE] - The error message here is clear, but consider including the actual values of the sides in the message to provide more context to the user.

Reply:
> [CODE STYLE] - The error message here is clear, but consider including the actual values of the sides in the message to provide more context to the user.

Good point about error messages